tests/internal/e2e/test/wdio.conf.ts

```diff
@@ -331,98 +331,99 @@ export const config: Options.Testrunner = {
-
     let checkVPython: string;
-    checkVPython =
-      process.platform == "win32" ? "where vpython" : "which vpython";
-
-    {
-      const { stdout, stderr } = await promisifiedExec(checkVPython);
-      if (stderr) {
-        console.log(stderr);
-        throw `Error when running ${checkVPython}`;
-      } else {
-        console.log(`vpython found in ${stdout}`);
-      }
-    }
-
     let checkClicast: string;
-    checkClicast =
-      process.platform == "win32" ? "where clicast" : "which clicast";
-
+    let vectorcastDir: string;
     let clicastExecutablePath: string;
-    {
-      const { stdout, stderr } = await promisifiedExec(checkClicast);
-      if (stderr) {
-        console.log(stderr);
-        throw `Error when running ${checkClicast}`;
-      } else {
-        clicastExecutablePath = stdout;
-        console.log(`clicast found in ${clicastExecutablePath}`);
-      }
-    }
-
-    process.env.CLICAST_PATH = clicastExecutablePath;
-
-    const vectorcastDir = path.dirname(clicastExecutablePath);
-    process.env.VC_DIR = vectorcastDir;
-
-    let clearScreenshots: string;
-    clearScreenshots =
-      process.platform == "win32" ? "del /s /q *.png" : "rm -rf *.png";
-    await promisifiedExec(clearScreenshots);
+    let testInputEnvPath: string;
+    let codedTestsPath: string;
 
     const testInputVcastTutorial = path.join(
       initialWorkdir,
       "test",
       "test_input",
       "vcastTutorial"
     );
-    const testInputEnvPath = path.join(testInputVcastTutorial, "cpp");
-    await mkdir(testInputEnvPath, { recursive: true });
 
-    const codedTestsPath = path.join(
-      testInputVcastTutorial,
-      "cpp",
-      "TestFiles"
-    );
-    await mkdir(codedTestsPath, { recursive: true });
-
-    const vscodeSettingsPath = path.join(testInputVcastTutorial, ".vscode");
-    await mkdir(vscodeSettingsPath, { recursive: true });
-    const launchJsonPath = path.join(vscodeSettingsPath, "launch.json");
-
-    let createLaunchJson: string;
-    createLaunchJson =
-      process.platform == "win32"
-        ? `copy /b NUL ${launchJsonPath}`
-        : `touch ${launchJsonPath}`;
-    await promisifiedExec(createLaunchJson);
-
-    const pathTovUnitInclude = path.join(vectorcastDir, "vunit", "include");
-    const c_cpp_properties = {
-      configurations: [
-        {
-          name: "Linux",
-          includePath: ["${workspaceFolder}/**", `${pathTovUnitInclude}`],
-          defines: [],
-          compilerPath: "/usr/bin/clang",
-          cStandard: "c17",
-          cppStandard: "c++14",
-          intelliSenseMode: "linux-clang-x64",
-        },
-      ],
-      version: 4,
-    };
-
-    const c_cpp_properties_JSON = JSON.stringify(c_cpp_properties, null, 4);
-    const c_cpp_properties_JSONPath = path.join(
-      vscodeSettingsPath,
-      "c_cpp_properties.json"
-    );
-    await writeFile(c_cpp_properties_JSONPath, c_cpp_properties_JSON);
+    if (process.env.VECTORCAST_DIR) {
```

Comment:
we have a lot of code duplication here. Could you figure out how to reduce the amount of duplicated code here? A lot of commands are exactly the same or called with a single parameter being calculated differently

Reply:
Refactored in 3c5f99218af5dc6b962549004871a5fe6788d65a.